docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+* <<data-deserialization>>
+* <<field-extraction>>
+* <<lookup-enrichment>>
+
+[[core-operations]]
+=== Performing Core Operations
+
+The plugins described in this section are useful for core operations, such as
+aggregating, mutating, and dropping events.
+
+Also see <<filter-plugins>> for the full list of available filter plugins. 
+
+<<plugins-filters-aggregate,aggregate filter>>::
+
+Aggregates information from several events (typically log lines) that belong to
+the same task and pushes the aggregated information into a final task event.
+For example:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  aggregate {
+    task_id => "%{country_name}"
+    code => "
+      map['country_name'] = event.get('country_name')
+      map['towns'] ||= []
+      map['towns'] << {'town_name' => event.get('town_name')}
+      event.cancel()
+    "
+    push_previous_map_as_event => true
+    timeout => 3
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-date,date filter>>::
+
+Parses dates from fields to use as Logstash timestamps for events.
++
+The following config parses a field called `logdate` to set the Logstash
+timestamp:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  date {
+    match => [ "logdate", "MMM dd yyyy HH:mm:ss" ]
+  }
+}   
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-drop,drop filter>>::
+
+Drops events. This filter is typically used in combination with conditionals.
++
+The following config drops 40% of the `debug` level log messages:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  if [loglevel] == "debug" {
+    drop {
+      percentage => 40
+    }
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-fingerprint,fingerprint filter>>::
+
+Fingerprints fields by applying a consistent hash.
++
+The following config fingerprints the `IP`, `@timestamp`, and `message` fields
+and adds the hash to a metadata field called `generated_id`:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  fingerprint {
+    source => ["IP", "@timestamp", "message"]
+    method => "SHA1"
+    key => "0123"
+    target => "[@metadata][generated_id]"
+  }
+}
+--------------------------------------------------------------------------------
+
+
+<<plugins-filters-mutate,mutate filter>>::
+
+Performs general mutations on fields. You can rename, remove, replace, and
+modify fields in your events.
++
+The following config renames the `HOSTORIP` field to `client_ip`:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  mutate {
+    rename => { "HOSTORIP" => "client_ip" }
```

Comment:
fixed - I've added strip, but not remove_field because that's actually one of the common filter options (I know...it's hard to tell...good thing we're fixing that)